app/webpacker/components/Schedule/VenuesAndRooms.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {
+  Form, Grid, Menu, Message,
+} from 'semantic-ui-react';
+import { getTextColor } from '../../lib/utils/calendar';
+import { toDegrees } from '../../lib/utils/edit-schedule';
+import i18n from '../../lib/i18n';
+
+export default function VenuesAndRooms({
+  venues,
+  activeVenueOrNull,
+  activeVenueIndex,
+  setActiveVenueIndex,
+  timeZoneCount,
+  rooms,
+  activeRoomIds,
+  dispatchRooms,
+}) {
+  const venueCount = venues.length;
+
+  const setActiveVenueIndexAndResetRooms = (newVenueIndex) => {
+    const newVenues = newVenueIndex > -1 ? [venues[newVenueIndex]] : venues;
+    const ids = newVenues.flatMap((venue) => venue.rooms).map((room) => room.id);
+    dispatchRooms({ type: 'reset', ids });
+
+    setActiveVenueIndex(newVenueIndex);
+  };
+
+  return (
+    <>
+      {venueCount > 1 && (
+        <Menu
+          pointing
+          secondary
+          fluid
+          stackable
+          // TODO: can't scroll left when 6+ venues
```

Comment:
This is unfortunately still a problem; see FMC 2024 (with 74 venues!) for example. I did try to fix this myself but got stuck. Someone more familiar with semantic ui might be able to easily fix it though.

![image](https://github.com/thewca/worldcubeassociation.org/assets/49137025/6864ee47-c346-41cc-bf07-2e97f3346940)